contracts/sentinel/guards/FuseWithdrawalGuard.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../IGuard.sol";
+import "../../refs/CoreRef.sol";
+import "../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+import "../../pcv/PCVGuardian.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+contract FuseWithdrawalGuard is IGuard, CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    struct WithdrawInfo {
+        address destination;
+        address underlying;
+        uint96 liquidityToLeave;
+    }
+
+    /// @notice map the destination and minimum liquidity for each pcv deposit
+    mapping(address => WithdrawInfo) public withdrawInfos;
+
+    EnumerableSet.AddressSet private fuseDeposits;
+
+    /// @notice the PCV mover contract exposed to guardian role
+    PCVGuardian public constant pcvGuardian = PCVGuardian(0x02435948F84d7465FB71dE45ABa6098Fc6eC2993);
+
+    /// @notice added to prevent dust from bricking the contract
+    uint256 public constant MIN_WITHDRAW = 100e18;
+
+    constructor(
+        address core,
+        address[] memory deposits,
+        address[] memory destinations,
+        address[] memory underlyings,
+        uint96[] memory liquidityToLeaveList
+    ) CoreRef(core) {
```

Comment:
Why is this coreRef? Guards ideally shouldn't rely on external state, which CoreRef brings in.